### ISSUE TITLE
New layout and updated BOM file

### ISSUE THREAD

**Issue by TurfControl** [valerionew/unisolder-notes#34]
Hello, I was in the market for an upgraded station at home and stumbled across this project.  I would like to do some design changes to make it easier to order mostly pick and placed.  I am not looking to take the design as my own or profit off of it, in fact I am more than willing to share all files 100% every step of the way.  I just finished the front board schematic into KI_CAD as it is my preferred software at work (only other option is PADS and I cant stand it anymore). if anyone would like to assist (and sparkybg approves) I would appreciate help finding new more available parts for the BOM.  if there are software designers out there interested in implementing any new features but require edits in the schematic i am happy to incorporate those edits.  I will be purchasing these from JLCPCB and just hand solder all the through hole components as required and will be more than happy to send off samples to anyone looking to add those software `integrations. 

my first step is going to be just making replica schematics so if anyone wants to double check the connections.  then I will begin to edit the schematic to relevant and more available parts either splitting diodes/op amps or joining them for example.  I will be changing the regulator to newer IC something most likely with a higher switching frequency i prefer TI based IC's so I can use the webench designer for simplicity. please any input or requests i am excited about this project, and if nobody else chimes in I will at least be laying out my own board 
Thank you 
Ryan

**Comment by valerionew** [comment]
Hey Ryan! 
One thing that i always wanted to do is porting the project from pic32 to stm32, as they are more and more spread in the industry. 
I think @sparkybg went for the pic32 as he had previous experience with it. 

However, with stm32 getting more and more prominent in the industry, it may be easier to port the project to that platform. Personally i won't have time, but there are a lot of devs out

**Comment by TurfControl** [comment]
That is something I can definitely implement.  Obviously I will need someone to work with on the project.  although I did take some programming classes in my past and I have written code for more than one micro (all 8 bit however), it is normally only for the proof of concept or testing my hardware. I can understand what many languages are doing but end up researching while writing it because I do not do enough of it.  With that being said I did forget to mention that I have a PM4 if there is anyone that wants to add hardware and write software for pic32 I can test and am willing to make more than 1 revision or dual footprint items, possibly make it more modular so that a daughter card can be used for testing hardware.

**Comment by sparkybg** [comment]
@valerionew You are right - STM32 is spread in the **industry**. It is a mess for DIY - there's at least 10 to 20 IDEs, with equal number of C compilers, the documentation is complete mess, compared with Microchip one, and you have a bucket full of programmer/debugger options, vs PicKit3 that can program and debug every chip that microchip is produced.

IMHO, PIC32 may not be the best MCU, but is HUGELY more efficient and less problematic to program for a DIYer. Same counts for nearly every other Microchip's MCU so far - you have to install MPLAB and XC32 (from a same site), buy PicKit3, route 5 wires, and start programming your first MCU.

I've tried Atmel (AVR, Attiny, 8051) , Renesas (RX62n), and some STMs - so I thisn I know what I am talking about.

Yes, it is frustrating not to have any PI32MXs on the market. But there's PIC32MK for example. I don't know if they are PIN compatible, but they have everything I've desired - 50% higher freq, FPU, 12bit ADC.

...and, there's already some clones with different routing, of course with problems. We have gain of 750 and microamp current sources - you have to be VERY understanding why something is routed the way it is in my project - 1mA current here and there, and you have more than enough offset or ground loop in order to produce absolutely unacceptable offsets in the end result, blaming the software or iron profiles for inconsistencies.

**Comment by TurfControl** [comment]
I completely understand there were difficulties and this board is definitely impressive.  I don't feel I am going to just throw something together all of the sudden and bam its better in every way....  

but I feel removing some of the options that you have implemented such as place the resistors for iron selections hard to the board and if interest increases for a new revision of the board I could include a jumper header selector with all resistors on the board. (unless someone solves this in software)

swapping out the 2675 for something much more efficient and higher frequency followed with an LDO on the output which will allow for removal of a lot of input capacitance. likewise for all used voltages.  (possibly separating all power conversion to a power distribution style board)

I have only worked on the schematics at this point and haven't done any true analysis to determine expected current draw in different locations of circuit at different voltages which will be necessary before selecting the IC.  I just finished the schematics but need to double check them a few times before working on the BOM and making any changes. so deep dive into the GBR's i go

**Comment by sparkybg** [comment]
1. Input capacitances are needed in order to have enough energy to save the options in the EEPROM when the controller is turned off. 2675 can work with much smaller capacitances already. :)

2. Higher frequency always means higher RFI to deal with.

3. 2675 is there mainly because of it's input voltage capability of 40V. most high frequency ones have far less max input voltage. This means linear regulator before them with much more disipation. The Darlington is already on the edge, so there must be a heatsink or something similar. And, the higher efficiency diminishes. :)

4. There is a simple (and cheap) negative supply circuit tied to it. Higher frequency devices in most cases means that this circuit must be made with separate switcher => more RFI and larger price.

When I designed this, I chose 2675 exactly because it was easy to obtain it from everywhere. There is no guarantee that the chip you use today will be available tomorrow.

Anyway, I am ready to discuss more detailed propositions. 

The circuit needs +3.3V 1A (it appears that 0.5A is not enough with some displays), -0.6V 0.1A, with separate wiring/filtering for display and for the rest of the circuit. The circuit also needs around 6V high impedance supply for valley voltage, and 10-15V supply for MOSFET gate drive. Also, as I already mentioned, the supply needs to have large enough capacitance at it's input in order to provide at least 0.3 - 0.5 seconds of stable 3.3V supply to the MCU and the display after the power is switched off.

**Comment by TurfControl** [comment]
That's all fantastic input thank you!

Higher frequencies always tend to have much more noise to go with them that will be a challenge I will have to deal with. I have had quite a bit of luck without even using LDO's on the output for proprietary wireless devices that I have in the field currently with the LMR16020 being a 60v input device capable of anywhere between 200k and 2.5Mhz switching allowing the use of a much smaller total footprint than the proven 2675. (i meant inductance earlier its the only drawback to the 2675)  This isn't a circuit I've even thought about yet as there are more than one voltage to consider as you mentioned. This IC is just one I'm familiar with in low noise required scenarios. Using these style devices can also require an extremely low esr capacitance meaning polymer caps which are on crazy shortage as well.

I haven't even begun making any schematic changes just double checking and developing a BOM while going over the GBR files.

**Comment by sparkybg** [comment]
Unfortunately, LMR16020 is as unavailable as 2675 at the moment. Or at least I cannot find it stocked anywhere.

It is a good chip, no doubt about it. A bit harder to solder also, because of the power pad. This is also one of the reasons I am using 2675 - it is plain SOIC8 without power pads and so on. Very easy to work with.

However, 60V part will make the Darlington preregulator's job a lot easier. And, 60->3.3V will work with extremely low duty and eficiency. 

As always, there's always a compromise that has to be done. :) And a good constructor is the one that knows how to make these. I am far from this - I tend to overprovision my circuits by a lot. :) But this either can be a good thing - as can be seen form the dangerous prototypes thread, overprovisioned DIY project can work even in an environment that it's constructor never considered. :) I had several "WTF" situations when I saw some implementations. And they work despite this. :)

**Comment by i2** [comment]
I am very much interested in this. Unfortunately, I can't be of help on the electronics side but I will follow this project passionately!

**Comment by TurfControl** [comment]
yeah the voltage drop from ~45-3.3v is quite intense off the bridge. 

I see a lot of stock on the IC on JLC or LCSC 7500 and the 3 amp part is expected this month for delivery of close to another 16k available.  using the IC would be similar to just the ability of a 2675hv just smaller inductor in value and size.

Another option i was considering was only stepping to a lower more efficient voltage with a 1 amp model then stepping down with LDO's for the necessary voltage requirements. in order to sustain 1A at 3.3V it would be difficult without external heatsinks so removing circuitry ends up adding other things meaning no space or BOM saving. But when i get around to deeper circuit analysis this is one area i was considering looking deeper into.

without a chip shortage it would be as simple as just grabbing a LMR36520 I haven't been able to get my hands on those since they were released and that has 70v transient safety with greater than 80% efficiency on a voltage variance of 45 input - output.

the compromise you are referencing is exactly why im open with my intentions. There is no part of this i would say i am an expert in.  I am happy with my talents but humble enough to admit there are many people smarter than i here.  hopefully with the people that followed this in the past or continue to build upon it but didn't want to invest the time of hardware design work will do so through me.

**Comment by valerionew** [comment]
Sorry to jump in without a thorough read of the past comments, I've only read them quickly. But i think there is an important point that i want to make:

If changes to the BOM have to be made, I think that the right direction is the one towards ease of sourcing. A more generic part with more pin-to-pin replacement for me will always be better than a better performing part with an unique package sold by only one seller. And this i think is particularly true for a DIY projects with such big number of parts.

So i think one of the objectives of the redesign could be this, together with reducing the part numbers count. I don't have the schematic in front of me right now, but i think there were some parts that can be swapped for others already in the BOM. 

What do you guys think?

**Comment by TurfControl** [comment]
Yes i completely agree with you on that one.  the regulator i was talking about earlier is actually quite popular but I don't know of any pin/pin compatible replacements from aftermarket sources where as the 2576 has been copied by everyone under the sun.  furthermore package changes can be made just from the last few years advancements in technology making things cheaper and just as available. 

if there are hardware revisions at all that anyone is interested in helping with before the board goes into proto phase i am happy to hear it.  

as I mentioned earlier the only thing I'm doing right now is creating the BOM with packages that are available specifically looking for ones available in china for pick and place use.  then I will do an initial layout

**Comment by sparkybg** [comment]
> i think one of the objectives of the redesign could be this, together with reducing the part numbers count. I don't have the schematic in front of me right now, but i think there were some parts that can be swapped for others already in the BOM.

You should be more specific on this. I can't remember of parts that cen be removed. In fact the last 5-6 revisions of the board was pretty much the opposite - you make the pcb, assemble it, and ten there's a problem and some parts need to be added to resolve it., or chematics/leyoutr need to be altered in some way.

**Comment by valerionew** [comment]
> You should be more specific on this. I can't remember of parts that cen be removed.

You're right! I promise that I'll check what my memory is vaguely remembering and get back to you! As soon as I finish my exam session (hopefully)
